File: blood-donor-system/src/main/java/com/blooddonor/blooddonorsystem/service/BloodStockService.java
package com.blooddonor.blooddonorsystem.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.blooddonor.blooddonorsystem.model.BloodStockStatus;
import com.blooddonor.blooddonorsystem.repository.BloodStockRepository;

@Service
public class BloodStockService {

    @Autowired
    private BloodStockRepository bloodStockRepository;

    public BloodStockStatus updateStock(BloodStockStatus stock) {
        return bloodStockRepository.save(stock);
    }

    public List<BloodStockStatus> getStockByRegion(String region) {
        return bloodStockRepository.findByRegion(region);
    }

    public List<BloodStockStatus> getAllStock() {
        return bloodStockRepository.findAll();
    }
}

File: blood-donor-system/frontend/src/services/bloodStockService.js
import api from "../api/axiosInstance";

const bloodStockService = {

    getAllStock: async () => {

        const response = await api.get("/stock");

        return response.data;

    },

    getStockByRegion: async (region) => {

        const response = await api.get(
            "/stock/region",
            { params: { region } }
        );

        return response.data;

    },

    updateStock: async (stockData) => {

        const response = await api.post(
            "/stock",
            stockData
        );

        return response.data;

    }

};

export default bloodStockService;
